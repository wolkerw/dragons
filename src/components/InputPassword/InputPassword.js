import React, { Component } from "react";
import PropTypes from "prop-types";
import "./InputPassword.scss";
import { Form } from "react-bootstrap";

class InputPassword extends Component {
    constructor(props) {
        super(props);
        this.state = {
            type: "password",
            firstInteraction: true,
            valid: false
        };
    }

    render() {
        const { placeholder } = this.props;

        const handleChange = event => {
            const value = event.target.value.length > 0 ? true : false;
            event.target.value = event.target.value.replace(/[\s]/g, "");
            this.setState({
                valid: value
            });
            this.props.validateCallback(event.target.value, "password", value);
        };

        const onBlur = () => {
            this.setState({ firstInteraction: false });
        };

        return (
            <div type="text" className="input-password" onBlur={onBlur}>
                <Form.Control
                    autoComplete="off"
                    name="password"
                    type={this.state.type}
                    placeholder={placeholder}
                    className={
                        "form-control" +
                        (this.props.firstSubmit &&
                        (this.state.firstInteraction || this.state.valid)
                            ? ""
                            : " is-invalid")
                    }
                    onChange={handleChange}
                />
            </div>
        );
    }
}

InputPassword.propTypes = {
    placeholder: PropTypes.string.isRequired,
    firstSubmit: PropTypes.bool
};
export default InputPassword;
