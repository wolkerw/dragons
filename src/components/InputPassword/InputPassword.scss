.input-password {
    input {
        padding: 0 54px;
        margin: 0 auto;
    }

    .password-eye {
        width: 40px;
        height: 40px;
        margin: -40px 0 0 280px;
        padding-top: 5px;
    }
}
