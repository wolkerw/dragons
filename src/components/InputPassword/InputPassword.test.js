import React from "react";
import InputPassword from "./InputPassword";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
// import messages from "../../assets/i18n";

configure({ adapter: new Adapter() });

const country = "pt-BR";
// const language = "pt-BR";
const placeholder = "Senha";
const callback = () => {
    console.log("TESTE");
};
const firstInteraction = true;

describe("InputPassword Component", () => {
    const component = (
        <IntlProvider
            locale={country}
            // key={country}
            // messages={messages[{ country }]}
            // defaultLocale={country}
        >
            <InputPassword
                placeholder={placeholder}
                validateCallback={callback}
                firstSubmit={firstInteraction}
            />
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("generated an input with the correct props", () => {
        const wrapper = mount(component);
        const props = wrapper.children().props();
        expect(props.placeholder).toEqual(placeholder);
        expect(props.firstSubmit).toEqual(firstInteraction);
        expect(props.validateCallback).toEqual(callback);
    });
});
