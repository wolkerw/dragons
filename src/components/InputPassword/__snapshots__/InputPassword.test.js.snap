// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword Component matches the snapshot 1`] = `
<div
  className="input-password"
  onBlur={[Function]}
  type="text"
>
  <input
    autoComplete="off"
    className="form-control form-control"
    name="password"
    onChange={[Function]}
    placeholder="Senha"
    type="password"
  />
</div>
`;
